cmd/antrea-agent/agent_windows.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"os/exec"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+// Initializer knows how to setup host the plugin for self-registration
+type ServiceInitializer struct {
+	serviceNameFlag  string
+}
```

Comment:
I think the registeration is simple enough and no real state needs to be kept via a struct so maybe just add a standalone function `registerServiceforOS` and call it directly (without initializing a struct first)

BTW, you need to define it in *_linux.go too, otherwise the compilation wouldn't pass.